### PR TITLE
Alert Monitor and handler

### PR DESCRIPTION
**Pull request by sebastianmueller01**

### CODE REVIEW

File: Core/Src/bmb.c [hunk 1]
```diff
@@ -460,35 +460,77 @@ void setGpio(uint32_t numBmbs, bool gpio0, bool gpio1, bool gpio2, bool gpio3)
...
+void aggregateBmbData(Bmb_S* bmb, uint32_t numBmbs)
 {
+	// Iterate through brick voltages
 	for (int i = 0; i < numBmbs; i++)
 	{
 		Bmb_S* pBmb = &bmb[i];
 		float maxBrickV = 0.0f;
 		float minBrickV = 5.0f;
 		float stackV	= 0.0f;
+		float maxBrickTemp = -200.0f;
```
Comment (sebastianmueller01) [inline Core/Src/bmb.c]: nix the magic numbers

File: Core/Src/bmb.c [hunk 2]
```diff
@@ -460,35 +460,77 @@ void setGpio(uint32_t numBmbs, bool gpio0, bool gpio1, bool gpio2, bool gpio3)
...
+		pBmb->maxBrickTemp = maxBrickTemp;
+		pBmb->minBrickTemp = minBrickTemp;
+		pBmb->avgBrickTemp = tempSum / NUM_BRICKS_PER_BMB;
+	}
+
+	// Iterate through board temperatures
+	for (int i = 0; i < numBmbs; i++)
+	{
+		Bmb_S* pBmb = &bmb[i];
+		float maxBoardTemp = -200.0f;
```
Comment (sebastianmueller01) [inline Core/Src/bmb.c:410]: nix magic numbers

File: Core/Src/alerts.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        // Determine if alert was cleared
+        if (checkTimerExpired(&alert->alertTimer))
+        {
+            // Timer expired - Clear alert
+            alert->alertStatus = ALERT_CLEARED;
+            // Load timer with alert set time
+            configureTimer(&alert->alertTimer, alert->setTime_MS);
+        }
+    }
```
Comment (sebastianmueller01) [inline Core/Src/alerts.c:62]: Handle else condition where alertStatus is neither set nor cleared. ( should never happen)

File: Core/Inc/timer.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+typedef struct
+{
+    uint32_t timCount;
+    uint32_t timThreshold;
+} Timer_S;
+
+void configureTimer(Timer_S* timer, uint32_t timerThreshold);
+
+void clearTimer(Timer_S* timer);
```
Comment (sebastianmueller01) [inline Core/Inc/timer.h:19]: change to HAL get tick

**Comment by sebastianmueller01** [comment]
We should try to verify the basic timing functionality. Try creating an alert that checks for a button press. That will allow us to verify the setting/clearing conditions and hysteresis etc. Have an LED or a print message correspond to the status of the alert

File: Core/Src/bms.c [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#include <bms.h>
+#include "bms.h"
 #include "bmb.h"
 
 Bms_S gBms;
```
Comment (isaiahwild42) [inline Core/Src/bms.c:4]: Define in maintask.c?

**Pull request merged**